phoenix-core/src/main/java/org/apache/phoenix/schema/PTableKey.java

```diff
@@ -26,7 +28,8 @@
     public PTableKey(PName tenantId, String name) {
         Preconditions.checkNotNull(name);
         this.tenantId = tenantId;
-        this.name = name;
+        this.name = !name.contains(QueryConstants.NAMESPACE_SEPARATOR) ? name
```

Comment:
I think we could potentially mask some bug or have it fail later than it should if we do this. We should remove this check altogether.